api/src/test/java/org/pmiops/workbench/reporting/ReportingUploadServiceTest.java

```diff
@@ -157,17 +197,54 @@ public void testUploadSnapshot_batchInserts() {
                 .fakeSize(4444L)
                 .creatorId(101L)));
 
-    reportingUploadService.uploadSnapshot(largeSnapshot);
+    reportingUploadServiceDmlImpl.uploadSnapshot(largeSnapshot);
     verify(mockBigQueryService, times(6))
         .executeQuery(queryJobConfigurationCaptor.capture(), anyLong());
 
     final List<QueryJobConfiguration> jobs = queryJobConfigurationCaptor.getAllValues();
     assertThat(jobs).hasSize(6);
-    final int researcherColumnCount = 4;
-    final int workspaceColumnCount = 5;
 
-    assertThat(jobs.get(0).getNamedParameters()).hasSize(researcherColumnCount * 5 + 1);
-    assertThat(jobs.get(4).getNamedParameters()).hasSize(researcherColumnCount + 1);
-    assertThat(jobs.get(5).getNamedParameters()).hasSize(workspaceColumnCount + 1);
+    assertThat(jobs.get(0).getNamedParameters()).hasSize(RESEARCHER_COLUMN_COUNT * 5 + 1);
+    assertThat(jobs.get(4).getNamedParameters()).hasSize(RESEARCHER_COLUMN_COUNT + 1);
+    assertThat(jobs.get(5).getNamedParameters()).hasSize(WORKSPACE_COLUMN_COUNT + 1);
+
+    final QueryParameterValue creationTime =
+        jobs.get(5).getNamedParameters().get("creation_time__0");
+    assertThat(creationTime).isNotNull();
+    final Instant instant = QueryParameterValues.timestampQpvToInstant(creationTime);
+    assertThat((double) instant.toEpochMilli()).isWithin(500.0).of(THEN.toEpochMilli());
+  }
+
+  @Test
+  public void testUploadSnapshot_streaming() {
+    final InsertAllResponse mockInsertAllResponse = mock(InsertAllResponse.class);
+    doReturn(Collections.emptyMap()).when(mockInsertAllResponse).getInsertErrors();
+
+    doReturn(mockInsertAllResponse)
+        .when(mockBigQueryService)
+        .insertAll(any(InsertAllRequest.class));
+    final ReportingJobResult result =
+        reportingUploadServiceStreamingImpl.uploadSnapshot(reportingSnapshot);
+    verify(mockBigQueryService, times(2)).insertAll(insertAllRequestCaptor.capture());
+    final List<InsertAllRequest> requests = insertAllRequestCaptor.getAllValues();
+
+    assertThat(requests).hasSize(2);
+
+    final List<RowToInsert> researcherRows = requests.get(0).getRows();
+    assertThat(researcherRows).hasSize(3);
+    assertThat(researcherRows.get(0).getId()).hasLength(16);
+    assertThat(researcherRows.get(0).getContent()).hasSize(RESEARCHER_COLUMN_COUNT + 1);
+
+    final List<RowToInsert> workspace1Rows = requests.get(1).getRows();
```

Comment:
this name is misleading - there is one row per workspace.  This is not Workspace 1's rows.

Please also assert that there are 3 workspaces here.